### PR TITLE
[Merged by Bors] - Separate collecting ATX deps from fetching

### PR DESCRIPTION
**Pull request by poszu**
## Motivation

Preparation for ATX V1 and V2 wire types

## Description

Separated collecting ATX deps (will be different for each ATX version) from fetching them (is generic).

## Test Plan

Adjusted unit tests

## TODO

- [x] Explain motivation or link existing issue(s)
- [x] Test changes and document test plan
- [ ] Update documentation as needed
- [ ] Update [changelog](../CHANGELOG.md) as needed

### CODE REVIEW

File: activation/handler.go [hunk 1]
```diff
@@ -647,38 +637,49 @@ func (h *Handler) processATX(
...
-			atx.ShortString(), atx.GetPoetProofRef().ShortString(), err,
-		)
-	}
-
-	atxIDs := make(map[types.ATXID]struct{}, 3)
-	if atx.PositioningATX != types.EmptyATXID && atx.PositioningATX != h.goldenATXID {
-		atxIDs[atx.PositioningATX] = struct{}{}
+// registerHashes registers that the given peer should be asked for
+// the hashes of the poet proof and ATXs.
+func (h *Handler) registerHashes(peer p2p.Peer, poetRef types.Hash32, atxIDs []types.ATXID) {
```
Comment (fasmat) [inline activation/handler.go:642]: Have we already decided if an ATX will be able to contain multiple `poetRef`s?

**Review by fasmat — APPROVED** [review]

File: activation/handler.go (hunk 1 above)
Comment (poszu) [reply activation/handler.go:642]: I think we did not. But if we decide so, this function can be adjusted to take a slice of poet-proof refs.

**Comment by poszu** [comment]
bors merge

**Comment by poszu** [comment]
bors merge

**Pull request closed without merging**